src/registry.service.ts

```diff
@@ -19,16 +20,31 @@ export class RegistryService {
   /**
    * Stores a directive into the registry.
    */
-  public add(arc: IArcHandler) { RegistryService.arcs.set(arc.getElement(), arc); }
+  public add(arc: IArcHandler) {
+    RegistryService.arcs.set(arc.getElement(), arc);
+    if (arc.exclude && arc.exclude()) {
+      this.excludedDeepCount++;
+    }
+  }
 
   /**
    * Removes a directive from the registry.
    */
-  public remove(arc: IArcHandler) { RegistryService.arcs.delete(arc.getElement()); }
+  public remove(arc: IArcHandler) {
+    RegistryService.arcs.delete(arc.getElement());
+    if (arc.exclude && arc.exclude()) {
+      this.excludedDeepCount--;
+    }
+  }
 
   /**
    * Returns the ArcDirective associated with the element. Returns
    * undefined if the element has no associated arc.
    */
   public find(el: HTMLElement) { return RegistryService.arcs.get(el); }
+
+  /**
+   * Returns whether there are any elements with deep exclusions in the registry.
+   */
+  public hasExcludedDeepElements(): boolean { return this.excludedDeepCount > 0; }
```

Comment:
In the current benchmark, findNextFocus takes 0.12ms of the scripting time, which is 0.02% of the benchmark time. I think the utility is worth a potential 0.005% decrease in performance.